src/FileDumper.cpp

```diff
@@ -67,28 +71,32 @@ void FileDumper::write(Frame& frame) {
         const auto subframe_data = frame.data() + off;
         const auto subframe_sz = std::min(config_.dump_frame, frame.size() - off);
 
-        write_subframe_(subframe_ts, subframe_data, subframe_sz);
+        write_subframe_(subframe_ts, subframe_data, subframe_sz, frame.get_IOType());
 
         off += subframe_sz;
     }
 }
 
-void FileDumper::write_subframe_(nanoseconds_t ts, const sample_t* buf, size_t bufsz) {
+  void FileDumper::write_subframe_(nanoseconds_t ts, const sample_t* buf, size_t bufsz, int iotype) {
     const int new_val
         = int(find_max(buf, bufsz) / config_.dump_rounding * config_.dump_rounding);
 
     const bool changed = (new_val != last_val_);
 
-    if (changed) {
+    if (changed && fp_ != stdout) {
         print_last_maybe_();
     }
+    else if (changed && iotype == 0){print_output_();}
+    else if (changed && iotype == 1){print_input_();}
```

Comment:
I kept the separate output formats so we don't have to change the python script

Reply:
> I kept the separate output formats so we don't have to change the python script

What is the rationale? I think that it's surprising and inconvenient for the user that changing dump destination would also change dump format.

Reply:
I was prioritizing the python script and I believed that if users weren't printing to stdout, they would choose sensible output file names to distinguish between the output dump and the input dump. If you still want to use the new dump format for all destinations, let me know so I can change it.

Reply:
I thought users would sensibly name their dump destinations to distinguish between input and output. If you want to use the new format, let me know so I can change it.

Reply:
@gavv do you want me to change it or anything else?

Reply:
Users may change destination here and there. Imagine you're debugging a shell pipeline that processes signal estimator output. One time you can redirect stdout to your pipeline, another time you can ask signal-estimator to save output to file and then feed the file to your pipeline. It would be inconvenient if your pipeline stops working and you need to adjust it depending on whether you're using files or pipes.

Another example. It would be surprising if these two commands will produce different results:

```
signal-estimator --dump-output "-" >file.txt
```

```
signal-estimator --dump-output file.txt
```

So yes, I suggest to unify the format,